pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -22,6 +22,9 @@ const (
 	// PausedAnnotation is the annotation that pauses the reconciliation (triggers
 	// an immediate requeue)
 	PausedAnnotation = "baremetalhost.metal3.io/paused"
+
+	// StatusAnnotation is the annotation that holds the Status of BMH
```

Comment:
It's not obvious why this is useful or how it gets used. I understand it from reading the bugzilla, but the behavior should also be documented here.

Reply:
Upstream uses it to hold the status of the CR while they pivot between clusters. We're using it for a different purpose. I suppose we could document it in some generic way upstream that makes sense downstream, too.

Reply:
Can you clarify what you're looking for? Adjusting the PR description? Code changes for the parameter documentation? Typically we just pull from upstream verbatim.

Reply:
Also, our use case is pretty much identical, we're pivoting the masters from the bootstrap to the cluster and want to preserve information learned in bootstrap to the cluster (introspection in our case, upstream is interested in more than that I realize)

Reply:
If I'm reading this to understand how to use the BareMetalHost API, this annotation is just not telling me how to use it. Why would you put the status in an annotation instead of in the Status? There's nothing here to tell me what exactly goes in this annotation, why it gets used, or what happens to the data that goes in it.

It would just help to describe that a JSON-serialized BareMetalHostStatus can be added here, that it'll be saved to the Status subresource by the controller, and that it's useful when host introspection was accomplished externally.

Reply:
We don't really want users to use this feature. The cluster-api system uses it when it pivots a host between clusters so that it can carry the inspection data. We're hacking the installer to do something similar. Users should not do that.

Reply:
I think these are all fine comments but it belongs upstream? I'm not sure I understand what the ask from me is on this PR.

Reply:
That's all fine. To me it seems worth documenting even if only for the sake of other metal3 developers (including me 6 months from now when I've forgotten what this does). It just seems like a gap to have part of the API undocumented, and the only way to find out what it does is read the code. I'm happy for that to happen upstream first if you are willing to accept this change here without documentation.

Reply:
I'm not disagreeing with the premise, this just isn't the right place. If you want to get it in upstream I'm happy to cherry-pick it as a separate BZ, but this is a tiny documentation nit against a severe bug that's been languishing unfixed for 3 OCP releases.

Reply:
I [filed an issue upstream](https://github.com/metal3-io/baremetal-operator/issues/524).